datalake/src/main/java/com/sequenceiq/datalake/service/rotation/CloudbreakPollerRotationExecutor.java

```diff
@@ -42,6 +42,16 @@ public void finalize(PollerRotationContext rotationContext) {
         sdxRotationService.rotateCloudbreakSecret(rotationContext.getResourceCrn(), (CloudbreakSecretType) rotationContext.getSecretType(), FINALIZE);
     }
 
+    @Override
+    public void preValidate(PollerRotationContext rotationContext) throws Exception {
+        sdxRotationService.checkFlowStateInCloudbreak(rotationContext.getResourceCrn());
```

Comment:
I think it would be better to use flowEndpoint.getLastFlowByResourceCrn

Reply:
discussed, will leave it empty